Content.Server/GameObjects/EntitySystems/ReactionSystem.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Content.Server.Chemistry;
+using Content.Server.GameObjects.Components.Chemistry;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.IoC;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.GameObjects.EntitySystems
+{
+    /// <summary>
+    /// Checks <see cref="SolutionComponent"/>s for conditions for chemical reactions and performs them if they're met.
+    /// </summary>
+    class ReactionSystem : EntitySystem
```

Comment:
As far as I can tell, your reactions happen instantly. Solutions are inert otherwise, so reactions can only really happen when you add a new reagent to the beaker.

As such, you can probably get rid of this system and make reactions checked only when reagents are added to this system, no?

Reply:
I thought about this overnight and agree. I can't think of any situation that this really *needs* to be a system. Even if we add temperature changes on solutions it shouldn't be necessary as long as we apply temperature changes through functions or a property of `SolutionComponent`. I'll move the systems code into `Solution` or `SolutionComponent`.

Reply:
You would need a system if you made reactions gradual in some way, but this currently isn't a thing so...

Reply:
One thing to note is that this change will require that functions that add reagents to solutions such as `SolutionComponent.TryAddReagent()` be moved from `Content.Shared` to `Content.Server` since reaction checks will be triggered in those, and reactions depend on types that are in `Content.Server`. They aren't used client-side at all, and I don't think clients should be able to directly change the contents of a solution anyways, so this shouldn't cause any issues.